File: 3364-minimum-positive-sum-subarray/3364-minimum-positive-sum-subarray.py
class Solution:
    def minimumSumSubarray(self, nums: List[int], l: int, r: int) -> int:
        n = len(nums)
        ans = float('inf')

        for sz in range(l, r + 1):
            curr = 0

            for i in range(sz):
                curr += nums[i]
            if curr > 0:
                ans = min(ans, curr)

            for i in range(sz, n):
                curr += nums[i] - nums[i - sz]
                if curr > 0:
                    ans = min(ans, curr)

        return -1 if ans == float('inf') else ans

File: 3364-minimum-positive-sum-subarray/3364-minimum-positive-sum-subarray.cpp
class Solution {
public:
    int minimumSumSubarray(vector<int>& nums, int l, int r) {
        int n = nums.size();
        int ans = INT_MAX;
        for (int sz = l; sz <= r; sz++) {
            int curr = 0;            
            for (int i = 0; i < sz; i++) curr += nums[i];
            if (curr > 0) ans = min(ans, curr);
            for (int i = sz; i < n; i++) {
                curr += nums[i] - nums[i - sz];                
                if (curr > 0) ans = min(ans, curr);
            }
        }
        return ans == INT_MAX ? -1 : ans;
    }
};